From foo@baz Fri Dec 18 11:04:54 PST 2009
Date: Fri, 18 Dec 2009 11:04:54 -0800
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: dt3155: add it to the build

Now it will build with the rest of the kernel

Cc: Scott Smedley <ss@aao.gov.au>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/Kconfig         |    2 ++
 drivers/staging/Makefile        |    1 +
 drivers/staging/dt3155/Kconfig  |    4 ++++
 drivers/staging/dt3155/Makefile |   33 ++++++---------------------------
 4 files changed, 13 insertions(+), 27 deletions(-)

--- /dev/null
+++ b/drivers/staging/dt3155/Kconfig
@@ -0,0 +1,4 @@
+config DT3155
+	tristate "DT3155 Digitizer support"
+	depends on PCI
+
--- a/drivers/staging/dt3155/Makefile
+++ b/drivers/staging/dt3155/Makefile
@@ -1,27 +1,6 @@
-
-ifeq ($(shell [[ `uname -r | cut -f 1,2 -d\.` < 2.6 ]] && echo pre2.6),pre2.6)
-# system with a pre 2.6 kernel _don't_ use kbuild.
-all:
-	$(MAKE) -f Makefile.pre-2.6
-
-clean:
-	rm -f *.o
-
-else
-# systems with a 2.6 or later kernel use kbuild.
-ifneq ($(KERNELRELEASE),)
-obj-m	:= dt3155.o
-dt3155-objs := dt3155_drv.o dt3155_isr.o dt3155_io.o allocator.o
-
-else
-KDIR	:= /lib/modules/$(shell uname -r)/build
-PWD		:= $(shell pwd)
-
-all:
-	$(MAKE) -C $(KDIR) M=$(PWD) modules
-
-clean:
-	rm -rf *.o *.mod *.mod.c *.ko  .dt3155* .allocator.o.cmd  .tmp_versions
-
-endif
-endif
+obj-$(CONFIG_DT3155)	+= dt3155.o
+dt3155-objs :=	\
+		dt3155_drv.o	\
+		dt3155_isr.o	\
+		dt3155_io.o	\
+		allocator.o
--- a/drivers/staging/Kconfig
+++ b/drivers/staging/Kconfig
@@ -143,5 +143,7 @@ source "drivers/staging/netwave/Kconfig"
 
 source "drivers/staging/sm7xx/Kconfig"
 
+source "drivers/staging/dt3155/Kconfig"
+
 endif # !STAGING_EXCLUDE_BUILD
 endif # STAGING
--- a/drivers/staging/Makefile
+++ b/drivers/staging/Makefile
@@ -53,3 +53,4 @@ obj-$(CONFIG_WAVELAN)		+= wavelan/
 obj-$(CONFIG_PCMCIA_WAVELAN)	+= wavelan/
 obj-$(CONFIG_PCMCIA_NETWAVE)	+= netwave/
 obj-$(CONFIG_FB_SM7XX)		+= sm7xx/
+obj-$(CONFIG_DT3155)		+= dt3155/
